x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/api/search_documents/search_documents_logic.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SearchResponseBody } from '@elastic/elasticsearch/lib/api/types';
+
+import { createApiLogic } from '../../../shared/api_logic/create_api_logic';
+import { HttpLogic } from '../../../shared/http';
+
+export const searchDocuments = async ({
+  indexName,
+  query,
+}: {
+  indexName: string;
+  query: string;
+}) => {
+  const route = `/internal/enterprise_search/search/${indexName}/${query}`;
```

Comment:
I'd still suggest adding `indices` in this URL. enterprise_search/search feels too generic, we're very specifically searching for documents in a single index.

Reply:
I see

Reply:
so `/internal/enterprise_search/indices/{indexName}/search/{query}`

this way it is under `indices` as you suggested and clear that we are searching in a single index. Not sure if it could cause issues with conflicts still

Reply:
Your first suggestion is perfect :)